src/test_utils/fixtures.ts

```diff
@@ -0,0 +1,118 @@
+import { Server } from 'node:http';
+import { fork } from 'node:child_process';
+import { AddressInfo } from 'node:net';
+import { join } from 'node:path';
+import { expect, test as originalTest } from 'vitest';
+import strip from 'strip-ansi';
+import emocks from 'emocks';
+import express from 'express';
+
+import { log } from '../log';
+
+type Fixtures = {
+  runMockServer: (mockDir: string) => Promise<Server>;
+  testCliCommand: (
+    args: string[],
+    expected?: {
+      snapshot?: true;
+      stderr?: string;
+      code?: number;
+    },
+  ) => Promise<void>;
+};
+
+export const test = originalTest.extend<Fixtures>({
+  // eslint-disable-next-line no-empty-pattern
+  runMockServer: async ({}, use) => {
+    let server: Server;
+    await use(async (mockDir) => {
+      const app = express();
+      app.use(express.json());
+      app.use(
+        '/',
+        emocks(join(process.cwd(), 'mocks', mockDir), {
+          '404': (_req, res) => res.status(404).send({ message: 'Not Found' }),
+        }),
+      );
+
+      await new Promise<void>((resolve) => {
+        server = app.listen(0, () => {
+          resolve();
+          log.debug(
+            'Mock server listening at %d',
+            (server.address() as AddressInfo).port,
+          );
+        });
+      });
+
+      return server;
+    });
+    await new Promise<void>((resolve, reject) =>
+      server.close((err) => (err ? reject(err) : resolve())),
+    );
+  },
+  testCliCommand: async ({ runMockServer, task }, use) => {
+    const mockDirOverride = /\*mockDir:(.*?)\*/.exec(task.name);
```

Comment:
why not just pass mockdir as other options via args?

Reply:
There was a reason behind that which didn't prove to be right :)